web/client/components/data/identify/viewers/JSONViewer.jsx

```diff
@@ -25,7 +25,7 @@ class JSONViewer extends React.Component {
         const RowViewer = (this.props.layer && this.props.layer.rowViewer) || this.props.rowViewer || PropertiesViewer;
         return (<div className="mapstore-json-viewer">
                 {(this.props.response.features || []).map((feature, i) => {
-                    return <RowViewer key={i} title={feature.id} exclude={["bbox"]} {...feature.properties}/>;
+                    return <RowViewer key={i} readOnly={feature.readOnly} title={feature.id} exclude={["bbox"]} {...feature.properties}/>;
```

Comment:
I think we should generalize this and pass down a more generic feature={feature}. For the future I would change the way properties from the features are passed down. I don't like the {...feature.properties} at all. For now, let's add feature as a property for the "full" object and let the annotations rowviewer use that to get the readonly value